odoo/service/model.py

```diff
@@ -72,17 +72,19 @@ def call_kw(model: BaseModel, name: str, args: list, kwargs: Mapping):
 
 
 def dispatch(method, params):
-    db, uid, passwd = params[0], int(params[1]), params[2]
+    db, uid, passwd, model, method_, *args = params
+    uid = int(uid)
     security.check(db, uid, passwd)
 
     threading.current_thread().dbname = db
     threading.current_thread().uid = uid
+    threading.current_thread().request_fragment = f'{model}/{method_}'
```

Comment:
Only need if you want to set it this early, otherwise `call_kw` would set it (see my other comment).